### ISSUE TITLE
virt-handler, node labeller: Improve modularity and testability

### ISSUE THREAD

**Issue by orelmisan** [kubevirt/kubevirt#12727]
**Is your feature request related to a problem? Please describe**:
This is a tracker issue for a long running effort.

Currently, the node labeller reads several files from the file system during its initialization:
https://github.com/kubevirt/kubevirt/blob/d1771b6fce43877db0cf65832b4e119ab06a3a84/pkg/virt-handler/node-labeller/node_labeller.go#L100

This fact implies that testing is performed by pointing Node labeller to read test data from the actual file system:
https://github.com/kubevirt/kubevirt/blob/d1771b6fce43877db0cf65832b4e119ab06a3a84/pkg/virt-handler/node-labeller/node_labeller_test.go#L56

Additionally, virt-handler's VM controller is directly dependent on the Node labeller:
https://github.com/kubevirt/kubevirt/blob/d1771b6fce43877db0cf65832b4e119ab06a3a84/cmd/virt-handler/virt-handler.go#L316

https://github.com/kubevirt/kubevirt/blob/d1771b6fce43877db0cf65832b4e119ab06a3a84/cmd/virt-handler/virt-handler.go#L333-L354

**Describe the solution you'd like**:
Node labeller should accept as input the minimal information it requires during initialization.
File system reads, parsing and data processing should be the caller's responsibility.

VM controller should be decoupled from Node labeller. 

**Describe alternatives you've considered**:

**Additional context**:
Please see PR #12652.

**Comment by orelmisan** [comment]
@iholder101 Per your suggestion, this issue will track our progress.
@Acedus FYI.

**Comment by anishbista60** [comment]
@orelmisan previoulsy,  i will work here. for the while, could you please look into`12662` and get it merged it's in final stage. Thankyou 
/assign

**Comment by iholder101** [comment]
> @iholder101 Per your suggestion, this issue will track our progress. @Acedus FYI.

Thanks for that!
I still think that a design proposal is better in this case since there we can discuss the overall design and plan moving forward instead of just outlining the general problem. This is not a must, but I think it will dramatically impact the amount of time that will be needed to review these changes.

**Comment by Acedus** [comment]
> > @iholder101 Per your suggestion, this issue will track our progress. @Acedus FYI.
> 
> Thanks for that!
> I still think that a design proposal is better in this case since there we can discuss the overall design and plan moving forward instead of just outlining the general problem. This is not a must, but I think it will dramatically impact the amount of time that will be needed to review these changes.

I agree. I've started working on a preliminary one, I'll be posting it soon for feedback.

**Comment by Acedus** [comment]
Wrote a draft design proposal: https://github.com/kubevirt/community/pull/322
FYI @orelmisan, @iholder101.

**Comment by iholder101** [comment]
Awesome @Acedus! Thank you!
